### PR TITLE
[aggregator] Fix width of string identifiers

### PR DESCRIPTION
**Pull request by georgemitenkov**
Makes width of string identifiers the same no matter what ID value is used, capped to 20 characters (maximum possible width of u64).

### CODE REVIEW

File: aptos-move/block-executor/src/executor.rs [hunk 1]
```diff
@@ -1387,12 +1387,19 @@ fn resource_group_error(err_msg: String) -> PanicOr<IntentionalFallbackToSequent
 }
 
 fn gen_id_start_value(sequential: bool) -> u32 {
-    // IDs are ephemeral. Pick a random prefix, and different each time,
-    // in case exchange is mistakenly not performed - to more easily catch it.
-    // And in a bad case where it happens in prod, to and make sure incorrect
-    // block doesn't get committed, but chain halts.
-    // (take a different range from parallel execution, to even more easily differentiate)
-
+    // All identifiers are ephemeral. We pick a fixed large value to start with
```
Comment (igor-aptos) [inline aptos-move/block-executor/src/executor.rs:1390]: This should never happen - and we need to make sure that is the case. 

But if it does - it is better to halt the chain, than store incorrect values. (Imagine random large identifier being stored as account balance). It is also easier to catch halting issues in testing, than incorrect value. 

Can you revert this change from this PR?

Comment (igor-aptos) [reply aptos-move/block-executor/src/executor.rs:1390]: Also making starting point fixed, doesn't make them deterministic,  as execution order in BlockSTM dictates which value gets which identifier. So this just makes it more likely to be identical in small tests, which is opposite of what we want

File: types/src/aggregator.rs [hunk 2]
```diff
@@ -83,7 +83,9 @@ impl TryIntoMoveValue for DelayedFieldID {
         Ok(match layout {
             MoveTypeLayout::U64 => Value::u64(self.as_u64()),
             MoveTypeLayout::U128 => Value::u128(self.as_u64() as u128),
-            layout if is_string_layout(layout) => bytes_to_string(to_utf8_bytes(self.as_u64())),
+            layout if is_string_layout(layout) => {
+                bytes_to_string(u64_to_fixed_size_utf8_bytes(self.as_u64()))
```
Comment (igor-aptos) [inline types/src/aggregator.rs:87]: can you add a comment here that we need to have a fixed sized identifier representation, to have consistent and deterministic gas charging?

**Review by igor-aptos — APPROVED** [review]

File: types/src/aggregator.rs [hunk 3]
```diff
@@ -164,8 +169,11 @@ fn string_to_bytes(value: Struct) -> Result<Vec<u8>, PanicError> {
         )
 }
 
-fn to_utf8_bytes(value: impl ToString) -> Vec<u8> {
-    value.to_string().into_bytes()
+fn u64_to_fixed_size_utf8_bytes(value: u64) -> Vec<u8> {
+    // Maximum u64 identifier size is 20 characters. We need a fixed size to
```
Comment (gelash) [inline types/src/aggregator.rs:173]: why is it 20 characters? can this somehow change (i.e. can we assert?)

**Review by gelash — APPROVED** [review]

File: types/src/aggregator.rs (hunk 3 above)
Comment (georgemitenkov) [reply types/src/aggregator.rs:173]: sorry, auto-merge😅 Because u64::max is 20 chars, this is what I meant in the comment, can update if unclear. The problem is that it we need to pass it to formatting macro and I was not able to pass a const expr there, but will double-check👍
